controllers/akodeploymentconfig/user/user_controller.go

```diff
@@ -310,8 +316,36 @@ func (r *AkoUserReconciler) getOrCreateAkoUserRole(roleTenantRef *string) (*mode
 		}
 		return r.aviClient.RoleCreate(role)
 	}
-	// else return role or error
-	return role, err
+	if err == nil {
+		return r.ensureAkoUserRole(role)
+	}
+	return nil, err
+}
+
+// ensureAkoUserRole ensure ako-essential-role has the latest permission
+func (r *AkoUserReconciler) ensureAkoUserRole(role *models.Role) (*models.Role, error) {
+	if role == nil {
+		// handle test panic
+		role = &models.Role{}
+		role, err := r.aviClient.RoleGetByName(akoov1alpha1.AkoUserRoleName)
+		if err != nil {
+			return role, err
+		}
+	}
+	// check if role needs to be sync
+	needSync := false
+	for i, permission := range role.Privileges {
```

Comment:
Is it better to generally check whether requested permissioin in role.Priviledges are presented, and update accordingly? Otherwise might need to modify this for loop apart from updating https://github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/commit/1385a26d9aa7960215ebec805f3c1f5dd694f77e

Reply:
> Is it better to generally check whether requested permission in role.Priviledges are presented, and update accordingly?

Actually those permissions are all presents, but it will show `NO_ACCESS` in type, so I am afraid that we have to find that permission and update its type.

Reply:
I assume you can still correct it if the privilege level is lower than we hardcoded in roles.Priviledges